compiler/z/codegen/OMRTreeEvaluator.cpp

```diff
@@ -2504,39 +2504,50 @@ genLoadAddressConstant(TR::CodeGenerator * cg, TR::Node * node, uintptr_t value,
       }
 
    TR::Compilation *comp = cg->comp();
-   cursor = generateRILInstruction(cg, comp->target().is64Bit() ? TR::InstOpCode::LLILF : TR::InstOpCode::IILF, node, targetRegister, static_cast<uint32_t>(value), cursor);
-
-   bool assumePatch = false;
-   bool isCompressedClassPointer = false;
-   if (node->isClassUnloadingConst())
+   if (node->isClassUnloadingConst() || comp->getOption(TR_EnableHCR))
```

Comment:
Now that we are revisiting this code, I want to know what are the cases where this genLoadAddressConstant is called and would need to be patched upon class redefinition and also cases which would not require patch and will generate instructions which won't be patched. 

Also, the consumer of this API in OpenJ9 where we would have `TR_EnableHCR` set by default, in that case (Unless we disable through option), would we ever execute the code added in this PR?